### PR TITLE
capture(cdc): fix cdc `campagin` stuck when send SIGSTOP to pd leader

### LINKED ISSUE #7470: cdc maybe stuck when send SIGSTOP to pd leader
### What did you do?

1. tiup start one cluster with two cdc & three cdc
2. `kill -19` to pd leader
3. `cdc cli capture list` can't list all capture, if `cdc cli capture list` sucessfully, repeat step 2.

### What did you expect to see?

`cdc cli capture list` list all capture

### What did you see instead?

`cdc cli capture list` can't list all capture

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
v6.3.0
```

Upstream TiKV version (execute `tikv-server --version`):

```console
v6.3.0 
```

TiCDC version (execute `cdc version`):

```console
v6.3.0
```

### PR DESCRIPTION
**Pull request by zeminzhou**
Signed-off-by: zeminzhou <zhouzemin@pingcap.com>

### What problem does this PR solve?

Issue Number: close #7470

### What is changed and how it works?
create etcd client in `capture` when `capture` reset

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 

##### Will it cause performance regression or break compatibility?
no
##### Do you need to update user documentation, design documentation or monitoring documentation?
no
### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Comment by zeminzhou** [comment]
PTAL~, thanks!

File: cdc/capture/capture.go [hunk 1]
```diff
@@ -332,7 +343,6 @@ func (c *captureImpl) run(stdCtx context.Context) error {
 			log.Info("campaign owner routine exited, restart the capture",
 				zap.String("captureID", c.info.ID))
 		}
```
Comment (asddongmen) [inline cdc/capture/capture.go:345]: why remove this log?

File: cdc/capture/capture.go [hunk 2]
```diff
@@ -465,35 +474,25 @@ func (c *captureImpl) campaignOwner(ctx cdcContext.Context) error {
...
-			}
-
-			log.Warn("owner resign timeout", zap.String("captureID", c.info.ID),
-				zap.Error(resignErr), zap.Int64("ownerRev", ownerRev))
-		}
-		cancel()
-
-		log.Info("owner resigned successfully",
-			zap.String("captureID", c.info.ID), zap.Int64("ownerRev", ownerRev))
+		// when owner exit, we don't `resign` because we will call `resign`
```
Comment (asddongmen) [inline cdc/capture/capture.go:477]: please remove this misleading log.

File: cdc/capture/capture.go [hunk 3]
```diff
@@ -465,35 +474,25 @@ func (c *captureImpl) campaignOwner(ctx cdcContext.Context) error {
...
 				zap.Error(err))
 			// for errors, return error and let capture exits or restart
 			return errors.Trace(err)
 		}
+
 		// if owner exits normally, continue the campaign loop and try to election owner again
 		log.Info("run owner exited normally",
 			zap.String("captureID", c.info.ID), zap.Int64("ownerRev", ownerRev))
+
+		// before a new cycle starts, we need resign and just need ignore error
```
Comment (asddongmen) [inline cdc/capture/capture.go:491]: Please add more comment about why we should just ignore this error.

File: cdc/capture/capture.go [hunk 4]
```diff
@@ -465,35 +474,25 @@ func (c *captureImpl) campaignOwner(ctx cdcContext.Context) error {
...
+		// when owner exit, we don't `resign` because we will call `resign`
+		// at `AsyncClose`
 		if err != nil {
 			log.Warn("run owner exited with error",
 				zap.String("captureID", c.info.ID), zap.Int64("ownerRev", ownerRev),
 				zap.Error(err))
 			// for errors, return error and let capture exits or restart
 			return errors.Trace(err)
 		}
+
```
Comment (3AceShowHand) [inline cdc/capture/capture.go:486]: please remove blank changes.

File: cdc/capture/capture_test.go [hunk 5]
```diff
@@ -70,6 +79,7 @@ func TestReset(t *testing.T) {
 		wg.Done()
 	}()
 	time.Sleep(100 * time.Millisecond)
+
```
Comment (3AceShowHand) [inline cdc/capture/capture_test.go:82]: please remove blank changes.

File: tests/mq_protocol_tests/framework/docker_compose_op.go [hunk 6]
```diff
@@ -134,6 +134,7 @@ func execInController(controller, shellCmd string) ([]byte, error) {
 	log.Info("Start executing in the Controller container",
 		zap.String("shellCmd", shellCmd), zap.String("container", controller))
 	cmd := exec.Command("docker", "exec", controller, "sh", "-c", shellCmd)
+
```
Comment (3AceShowHand) [inline tests/mq_protocol_tests/framework/docker_compose_op.go:137]: please remove blank changes.

**Review by asddongmen — APPROVED** [review]

**Review by 3AceShowHand — APPROVED** [review]

File: cdc/api/status/status.go [hunk 7]
```diff
@@ -63,7 +63,12 @@ func (h *statusAPI) handleDebugInfo(w http.ResponseWriter, req *http.Request) {
 	ctx := req.Context()
 	h.capture.WriteDebugInfo(ctx, w)
 	fmt.Fprintf(w, "\n\n*** etcd info ***:\n\n")
-	h.writeEtcdInfo(ctx, h.capture.GetEtcdClient(), w)
+	etcdClient, err := h.capture.GetEtcdClient()
+	if err != nil {
+		fmt.Fprintf(w, "failed to etc client: %s\n\n", err.Error())
```
[the remark this answers is not in the document]
Comment (zeminzhou) [reply cdc/api/status/status.go]: fixed, thanks

File: cdc/capture/capture.go (hunk 1 above)
Comment (zeminzhou) [reply cdc/capture/capture.go]: wrong deletion, add back, PTAL~

File: cdc/capture/capture.go (hunk 2 above)
Comment (zeminzhou) [reply cdc/capture/capture.go]: removed

File: cdc/capture/capture.go (hunk 4 above)
Comment (zeminzhou) [reply cdc/capture/capture.go]: removed, thanks!

File: cdc/capture/capture_test.go (hunk 5 above)
Comment (zeminzhou) [reply cdc/capture/capture_test.go]: removed, thanks!

File: tests/mq_protocol_tests/framework/docker_compose_op.go (hunk 6 above)
Comment (zeminzhou) [reply tests/mq_protocol_tests/framework/docker_compose_op.go]: removed, thanks!

File: cdc/capture/capture.go (hunk 3 above)
Comment (zeminzhou) [reply cdc/capture/capture.go]: the best way should return error, thanks!

**Pull request merged**